src/QMCHamiltonians/BareKineticEnergy.cpp

```diff
@@ -272,24 +280,61 @@ void BareKineticEnergy::evaluateOneBodyOpMatrix(ParticleSet& P,
     M.push_back(zeromat);
     grad_M.push_back(zerogradmat);
     lapl_M.push_back(zeromat);
+    gradJdotgradPhi.push_back(zeromat);
   }
 
   psi.getEGradELaplM(P, M, grad_M, lapl_M);
+  psi.evaluateJastrowVGL(P, G, L);
+
   for (int ig = 0; ig < ngroups; ig++)
   {
-    const IndexType sid = psi.getTWFGroupIndex(ig);
-    lapl_M[ig] *= MinusOver2M[ig];
-    B[sid] += lapl_M[ig];
+    const IndexType sid    = psi.getTWFGroupIndex(ig);
+    const IndexType norbs  = psi.numOrbitals(sid);
+    const IndexType first  = P.first(ig);
+    const IndexType last   = P.last(ig);
+    const IndexType nptcls = last - first;
+    for (int iel = first; iel < last; iel++)
+    {
+      for (int iorb = 0; iorb < norbs; iorb++)
+      {
+        gradJdotgradPhi[sid][iel - first][iorb] = 2.0 * dot(G[iel], grad_M[sid][iel - first][iorb]);
+        B[sid][iel - first][iorb] += MinusOver2M[ig] *
+            (lapl_M[sid][iel - first][iorb] + gradJdotgradPhi[sid][iel - first][iorb] +
+             (L[iel] + dot(G[iel], G[iel])) * M[sid][iel - first][iorb]);
+      }
+    }
   }
 }
 
 void BareKineticEnergy::evaluateOneBodyOpMatrixForceDeriv(ParticleSet& P,
-                                                          const ParticleSet& source,
+                                                          ParticleSet& source,
                                                           const TWFFastDerivWrapper& psi,
                                                           const int iat,
                                                           std::vector<std::vector<ValueMatrix>>& Bforce)
 {
   IndexType ngroups = P.groups();
+  IndexType nelec   = P.getTotalNum();
+
+  ParticleSet::ParticleGradient Gtmp, G;
+  ParticleSet::ParticleLaplacian Ltmp, L;
+  Gtmp.resize(nelec);
+  G.resize(nelec);
+  Ltmp.resize(nelec);
+  L.resize(nelec);
+
+  std::vector<ValueMatrix> M;
+  std::vector<GradMatrix> grad_M;
+  std::vector<ValueMatrix> lapl_M;
```

Comment:
Could you do a printf check to see if the tests hit here? Codecov says no but it is doubtful.

Reply:
Print statement check works.  This is weird, because test_ion_deriv literally calls this function and a "pass" requires the output from this function to be correct.